File: Python3/프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    pattern1 = [1,2,3,4,5]
    pattern2 = [2,1,2,3,2,4,2,5]
    pattern3 = [3,3,1,1,2,2,4,4,5,5]
    
    answer1 = 0
    answer2 = 0
    answer3 = 0
    
    
    for i,a in enumerate(answers):
        if a == pattern1[i % 5]:
            answer1 += 1
        
        if a == pattern2[i % 8]:
            answer2 += 1
        
        if a == pattern3[i % 10]:
            answer3 += 1
    
    res = [answer1, answer2, answer3]
    
    max_value = max(res)
    answer = []
    
    for i,a in enumerate(res):
        if a == max_value:
            answer.append(i+1)
            
    return answer
    
            
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
    
#     num1 = [1,2,3,4,5]
#     num2 = [2,1,2,3,2,4,2,5]
#     num3 = [3,3,1,1,2,2,4,4,5,5]
    
#     result = [0 for _ in answers]

#     for i,answer in enumerate(answers):
#         if num1[(i + 5) % 5] == answer:
#             result[0] += 1
            
#         if num2[(i + 8) % 8] == answer:
#             result[1] += 1
    
#         if num3[(i + 10) % 10] == answer:
#             result[2] += 1
    
#     max_value = max(result)
#     answer = []
    
#     for i,res in enumerate(result):
#         if res == max_value:
#             answer.append(i+1)
    
#     return answer







# def solution(answers):
#     answer = []
#     first = [1,2,3,4,5]
#     second = [2,1,2,3,2,4,2,5]
#     third = [3,3,1,1,2,2,4,4,5,5]
#     result = []
    
#     for lst in [first, second, third]:
#         cnt = 0
#         temp = 0
#         for ans in answers:
#             if lst[cnt] == ans:
#                 temp += 1
#             cnt += 1
#             if cnt == len(lst):
#                 cnt = 0
            
#         result.append(temp)
    
#     max_value = max(result)
    
#     for id in range(len(result)):
#         if result[id] == max_value:
#             answer.append(id+1)
            
#     return answer

File: C++/프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> s1 = {1,2,3,4,5};
    vector<int> s2 = {2,1,2,3,2,4,2,5};
    vector<int> s3 = {3,3,1,1,2,2,4,4,5,5};
    
    int a1 = 0;
    int a2 = 0;
    int a3 = 0;
    
    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == s1[i % 5]) a1++;
        if (answers[i] == s2[i % 8]) a2++;
        if (answers[i] == s3[i % 10]) a3++;
    }
    
    int max_value = max({a1, a2, a3});
    if (a1 ==max_value) answer.push_back(1);
    if (a2 == max_value) answer.push_back(2);
    if (a3 == max_value) answer.push_back(3);
    return answer;
}